### DISCUSSION TITLE
Creating Stream Provider (not StreamProvider) with riverpod generator

### DISCUSSION THREAD

**Discussion by urusaich** [rrousselGit/riverpod#3195] [Q&A]
Hi! I am exploring generator and faced a problem. 
I need to generate a Provider which returns Stream object, but when i am trying to do it, generator created a StreamProvider with AsyncValue data. 

This code:
```dart
@Riverpod(keepAlive: true)
Stream<dynamic> stream(StreamRef ref) => const Stream.empty();
```

Generates:
```dart
StreamProvider<dynamic>
```

What i expect:

```dart
Provider<Stream<dynamic>>
```

I need a broadcast stream for use it by different components
Any thoughts on what i should do in this situation?
Thanks

**Comment by rrousselGit** [comment]
See https://riverpod.dev/docs/essentials/websockets_sync#disabling-conversion-of-streamsfutures-to-asyncvalue

**Reply by urusaich** [reply]
Oh, thanks!
